docs-next/pages/apis/context.mdx

```diff
@@ -2,6 +2,139 @@ import { Markdown } from '../../components/Page';
+  session,
+  startSession,
+  endSession,
+
+  // New context creators
+  sudo,
+  exitSudo,
+  withSession,
+
+  // Database access
+  knex,
+  prisma,
+  mongoose,
+
+  // Access control helpers
+  getListAccessControlForUser,
+  getFieldAccessControlForUser,
+
+  // Internal state
+  totalResults,
+  maxTotalResults,
+  schemaName,
+
+  // Deprecated
+  gqlNames,
+  executeGraphQL,
+  keystone,
+};
+```
+
+### HTTP request object
+
+`req`: The [IncomingMessage](https://nodejs.org/api/http.html#http_class_http_incomingmessage) object from the HTTP request which called the GraphQL API.
+
+### List item API
+
+`lists`: The [list items API](./list-items), which can be used to perform CRUD operations against your GraphQL API.
+
+### GraphQL helpers
+
+`graphql.schema`: The [GraphQL Schema](https://graphql.org/graphql-js/type/#graphqlschema) object, which can be used directly, or via `graphql.raw`, `graphql.run`, or the `lists` API.
+
+`graphql.raw`: An async function which takes `({ query, variables })` and executes the query against the GraphQL schema, returning `{ data, errors }`.
+`query` can be either a string or a GraphQL [`Document`](https://graphql.org/graphql-js/language/#parse).
+`variables` is an optional object containing the input variables for the query.
+The returned object is an [`ExecutionResult`](https://graphql.org/graphql-js/execution/).
+This query uses the current `context` object as its `context`.
+
+`graphql.run`: An async function which takes `({ query, variables })` and executes the query against the GraphQL schema, returning `data`.
+`query` can be either a string or a GraphQL [`Document`](https://graphql.org/graphql-js/language/#parse).
+`variables` is an optional object containing the input variables for the query.
+The returned object is an [`ExecutionResult.data`](https://graphql.org/graphql-js/execution/).
+Any errors generated by the query will be thrown as an exception.
+This query uses the current `context` object as its `context`.
+
+### Session API
+
+If you configure your Keystone system with session management then you will have access to the following properties.
+See the [session API](./session#session-context) for more details.
+
+`session`: The current session data object.
+
+`startSession`: An internal helper function used by authentication mutations to start a session on a successful login. This should not be called directly.
+
+`endSession`: An internal helper function used by authentication mutations to end a session on logout. This should not be called directly.
+
+### New context creators
+
+When using the `context.lists`, `context.graphql.run`, and `context.graphql.raw` APIs, access control and session information is passed through to these calls from the `context` object.
+The following functions will create a new `KeystoneContext` object with this behaviour modified.
+
+`context.sudo()` returns a new `KeystoneContext` object with all access control disabled for subsequent API calls.
```

Comment:
It looks like you're switching between `fnName` and `context.fnName` here and in the Session API section above ..? I'm not sure which one is a better choice, but this reads like they're different, and I don't think they are (?)

Reply:
Yeah... I'm also 50/50 on this... let me have a think about it 🤔